cardano-crypto-class/src/Cardano/Crypto/Seed.hs

```diff
@@ -139,7 +144,13 @@ getRandomBytesFromSeed n =
           Identity $
             getBytesFromSeed (fromIntegral n) s
 
-
 instance MonadRandom MonadRandomFromSeed where
   getRandomBytes n = BA.convert <$> getRandomBytesFromSeed n
 
+-- Libsodium co-op
+
+mlsbToSeed :: KnownNat n => NaCl.MLockedSizedBytes n -> Seed
+mlsbToSeed = mkSeedFromBytes . NaCl.mlsbToByteString
```

Comment:
Do we need to warn about losing mlock-guarantees here?

Reply:
Yes, `mlsbToByteString` comment apply here as well. OTOH, this is used only in tests, and `KES.Mock` and `KES.Simple` so I'm even tempted to mark those (`mlsbToSeed` and `mlsbToByteString`) as `{-# DEPRECATED #-}` with a message "do not use" (c.f. `traceMLockedForeignPtr`).

Reply:
We could instead parametrise the KES class over its seed type. Then in the testing instances, we instantiate it to `Seed`, and to `MlockedSizedBytes` in the real instances. No need to have any means to convert between them

Reply:
`mlsbFromSeed` is used only in tests. I will remove it from there.

Reply:
I'm against parametrising KES type over seed type. That would be used only for mock (and maybe simple?) variants, which are not used in "production".

IMHO it's unnecessary degree of freedom. (Similarly as every DSIGN instance has `type Signable MockDSIGN = SignableRepresentation` which only makes type-signatures uglier and more work to type-checker).

Reply:
That's not quite true, there is a DSIGN instance in the Byron ledger spec which has a different instance (which is precisely the reason for that freedom!)

Reply:
Ok.

Yet, `Seed` cannot be used. It have to be type which tells it's size on the type level, so it could be some `SizedSeed (n :: Nat)` (which can be backed up by non-mlocked `PinnedSizedBytes`).

I have since removed `mlsbToSeed`. It's better to figure out a type-class for seeds in a follow up. One would need some functions to facilitate serialization, I don't want to linger with this PR any further.